packages/phone-number-privacy/combiner/src/common/handlers.ts

```diff
@@ -90,9 +130,98 @@ export function meteringHandler<R extends OdisRequest>(
   }
 }
 
+export function timeoutHandler<R extends OdisRequest>(
+  timeoutMs: number,
+  handler: PromiseHandler<R>
+): PromiseHandler<R> {
+  return async (req, res) => {
+    const timeoutSignal = (AbortSignal as any).timeout(timeoutMs)
+    timeoutSignal.addEventListener(
+      'abort',
+      () => {
+        if (!res.headersSent) {
+          sendFailure(ErrorMessage.TIMEOUT_FROM_SIGNER, 500, res, req.url)
+        }
+      },
+      { once: true }
+    )
+
+    await handler(req, res)
+  }
+}
+
+export function withEnableHandler<R extends OdisRequest>(
+  enabled: boolean,
+  handler: PromiseHandler<R>
+): PromiseHandler<R> {
+  return async (req, res) => {
+    if (enabled) {
+      return handler(req, res)
+    } else {
+      sendFailure(WarningMessage.API_UNAVAILABLE, 503, res, req.url)
+    }
+  }
+}
+
 export async function disabledHandler<R extends OdisRequest>(
-  _: Request<{}, {}, R>,
+  req: Request<{}, {}, R>,
   response: Response<OdisResponse<R>, Locals>
 ): Promise<void> {
-  sendFailure(WarningMessage.API_UNAVAILABLE, 503, response)
+  sendFailure(WarningMessage.API_UNAVAILABLE, 503, response, req.url)
+}
+
+export function sendFailure(
+  error: ErrorType,
+  status: number,
+  response: Response,
+  _endpoint: string,
```

Comment:
`_endpoint` is not used.

should it be made optional and changed to `endpoint`

Reply:
I left that like that because it was used by the counters which I removed but maybe was something we were going to add later.
The `_xxx` is to avoid a check of usage. If I set it to optional, it will still fail because it's not used inside.
I can remove that, but as it is, it will make our life easier for the future refactor of the commons